cmd/plume/prerelease.go

```diff
@@ -0,0 +1,233 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"compress/bzip2"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/Microsoft/azure-vhd-utils-for-go/vhdcore/validator"
+	"github.com/spf13/cobra"
+	"golang.org/x/net/context"
+
+	"github.com/coreos/mantle/auth"
+	"github.com/coreos/mantle/platform/api/azure"
+	"github.com/coreos/mantle/sdk"
+	"github.com/coreos/mantle/storage"
+)
+
+var (
+	preReleaseDryRun bool
+	cmdPreRelease    = &cobra.Command{
+		Use:   "pre-release [options]",
+		Short: "Run pre-release steps for CoreOS",
+		Long:  "Runs pre-release steps for CoreOS, such as image uploading and OS image creation, and replication across regions.",
+		RunE:  runPreRelease,
+	}
+
+	azureOpts = azure.Options{}
+)
+
+func init() {
+	cmdPreRelease.Flags().StringVar(&azureOpts.PublishSettingsFile, "publish-settings", "", "publish settings file")
+	cmdPreRelease.Flags().BoolVarP(&preReleaseDryRun, "dry-run", "n", false,
+		"perform a trial run, do not make changes")
+
+	AddSpecFlags(cmdPreRelease.Flags())
+	root.AddCommand(cmdPreRelease)
+}
+
+func runPreRelease(cmd *cobra.Command, args []string) error {
+	if len(args) > 0 {
+		return errors.New("no args accepted")
+	}
+
+	spec := ChannelSpec()
+	ctx := context.Background()
+	client, err := auth.GoogleClient()
+	if err != nil {
+		return err
+	}
+
+	src, err := storage.NewBucket(client, spec.SourceURL())
+	if err != nil {
+		return err
+	}
+
+	// Sanity check!
+	if vertxt := src.Object(src.Prefix() + "version.txt"); vertxt == nil {
```

Comment:
I get that, but what is the intent? The contents are never used. I assume it's to make sure that the version exists in the bucket, but why not just let it 404 when pulling the image?